nodes/list_struct/shift_mk2.py

```diff
@@ -66,8 +66,16 @@ def process(self):
             return
 
         data = self.inputs['data'].sv_get(deepcopy=False)
-        number = self.inputs["shift"].sv_get(deepcopy=False)[0][0]
+        number = self.inputs["shift"].sv_get(deepcopy=False)
+        output = []
+        for d,n in zip(data,number):
+            out = self.new_method(d, n[0])
+            output.append(out)
 
+
+        self.outputs['data'].sv_set(output)
+
+    def new_method(self, data, number):
```

Comment:
The name of the new method should be something more meaningful. Something like roll list.